app/services/data_sources/json_environment_variable_data_source.rb

```diff
@@ -21,6 +21,9 @@ class << self
     end
 
     # can't have us reading and writing to a file at the same time
+    # TODO: currently we just have this global lock
```

Comment:
💯

Reply:
I actually created an issue for this also https://github.com/fastlane/ci/issues/771